catalogue/tests/test_config.py

```diff
@@ -0,0 +1,1428 @@
+import inspect
+
+import pytest
+from typing import Dict, Optional, Iterable, Callable, Any, Union
+from types import GeneratorType
+from pydantic import BaseModel, StrictFloat, PositiveInt, constr
+from pydantic.types import StrictBool
+
+from catalogue import ConfigValidationError, Config
+from catalogue.config.util import make_tempdir, Generator, partial
+from catalogue.tests.util import Cat, my_registry
+import numpy
+import pickle
+
+
+EXAMPLE_CONFIG = """
```

Comment:
This `EXAMPLE_CONFIG` seems quite elaborate for really just a small quick test - perhaps we can simplify this and make it look less like a spaCy config (just to be more generic)?